x/incentives/keeper/distribute_test.go

```diff
@@ -1376,3 +1451,261 @@ func (s *KeeperTestSuite) TestCreateGroupGaugeAndDistribute() {
+			GaugeId:          record.GaugeId,
+			CurrentWeight:    record.CurrentWeight,
+			CumulativeWeight: record.CumulativeWeight,
+		}
+	}
+
+	return types.GroupGauge{
+		GroupGaugeId: src.GroupGaugeId,
+		InternalGaugeInfo: types.InternalGaugeInfo{
+			TotalWeight:  src.InternalGaugeInfo.TotalWeight,
+			GaugeRecords: gaugeRecords,
+		},
+		SplittingPolicy: src.SplittingPolicy,
+	}
+}
+
+// withUpdatedVolumes takes in a group gauge and a list of updated cumulative volumes (ordered) and updates the contents of the gauge to
+// reflect these new volumes.
+// It is only intended to be used to set expected values for test cases.
+func (s *KeeperTestSuite) withUpdatedVolumes(groupGauge types.GroupGauge, updatedCumulativeVolumes []osmomath.Int) types.GroupGauge {
+	// Ensure there aren't more volumes to update than records in group gauge
+	s.Require().True(len(updatedCumulativeVolumes) <= len(groupGauge.InternalGaugeInfo.GaugeRecords))
+
+	// We make a deep copy of the group gauge to ensure we don't modify the original input/defaults
+	updatedGroupGauge := deepCopyGroupGauge(groupGauge)
+
+	newTotalWeight := osmomath.ZeroInt()
+	for i, updatedVolume := range updatedCumulativeVolumes {
+		currentRecord := groupGauge.InternalGaugeInfo.GaugeRecords[i]
+		updatedRecord := types.InternalGaugeRecord{
+			GaugeId:          currentRecord.GaugeId,
+			CurrentWeight:    updatedVolume.Sub(currentRecord.CumulativeWeight),
+			CumulativeWeight: updatedVolume,
+		}
+		updatedGroupGauge.InternalGaugeInfo.GaugeRecords[i] = updatedRecord
+		newTotalWeight = newTotalWeight.Add(updatedRecord.CurrentWeight)
+	}
+	updatedGroupGauge.InternalGaugeInfo.TotalWeight = newTotalWeight
+
+	return updatedGroupGauge
+}
+
+// withSplittingPolicy returns a deep copy of the passed in group gauge with the splitting policy set to the passed in value.
+func withSplittingPolicy(groupGauge types.GroupGauge, splittingPolicy types.SplittingPolicy) types.GroupGauge {
+	// We make a deep copy of the group gauge to ensure we don't modify the original input/defaults
+	updatedGroupGauge := deepCopyGroupGauge(groupGauge)
+	updatedGroupGauge.SplittingPolicy = splittingPolicy
+
+	return updatedGroupGauge
+}
+
+// setPoolVolumes takes in an array of pool IDs and volumes and sets each pool's volume to the corresponding volume amount.
+// If there are more pool IDs than volumes, the extra pool IDs are ignored. This is to more simply accommodate cases where only the first k
+// of n pools are updated without needing to pad the volumes array during test setup.
+func (s *KeeperTestSuite) setPoolVolumes(poolIds []uint64, volumes []osmomath.Int) {
+	s.Require().True(len(poolIds) >= len(volumes))
+
+	for i, curVolume := range volumes {
+		s.App.PoolManagerKeeper.SetVolume(s.Ctx, poolIds[i], sdk.NewCoins(sdk.NewCoin(s.App.StakingKeeper.BondDenom(s.Ctx), curVolume)))
+	}
+}
+
+// TODO: rename this to syncVolumeSplitGroup as part of https://github.com/osmosis-labs/osmosis/pull/6446
```

Comment:
These new TODOs have been addressed (and subsequently removed) in the downstream PR